README.md

```diff
@@ -211,49 +224,63 @@ To fetch git submodules in repositories set `submodules`:
 }
 ```
 
-<!--
-This currently does not work as advertised at least for modules
+## NixOS modules, overlays and library function support
 
-## Conventions for NixOS modules, overlays and library functions
+It is also possible to define more than just packages. In fact any Nix expression can be used.
 
 To make NixOS modules, overlays and library functions more discoverable,
 we propose to put them in their own namespace within the repository.
 This allows us to make them later searchable, when the indexer is ready.
 
-Put all NixOS modules in the `modules` attribute of your repository:
+### Usage on NixOS
+
+If you intend to use modules, overlays and library functions in your NixOS configuration.nix, you need to take care to not introduce infinite recursion. Specifically, you need to import NUR like this in the modules:
 
 ```nix
-# default.nix
-{
-  modules = ./import modules;
+{ pkgs, config, lib, ... }:
+let
+  nur = import (builtins.fetchTarball "https://github.com/nix-community/NUR/archive/master.tar.gz") {};
```

Comment:
Hmm, I'm not sure, because `pkgs` isn't actually needed when you just use NUR for modules/overlays/library functions.

Oh I think what would be neat is to change the default `pkgs` argument used by NUR from `import <nixpkgs> {}` to `throw "NUR didn't receive a pkgs argument"`. Then usage without a `pkgs` argument is just fine as long as you don't evaluate it (which is the case when you only need NixOS modules), but as soon as you try to use a package, you get the throw.